backend/app/controllers/api/v1/restaurants_controller.rb

```diff
@@ -0,0 +1,85 @@
+module Api
+  module V1
+    class RestaurantsController < ApplicationController
+      before_action :set_restaurant, only: [:show, :update, :destroy]
+
+      # GET /restaurants
+      def index
```

Comment:
I'm generally  fan of small code blocks, but in particular when it's in your system boundary.
Controllers should preferably be slim in code and logic.
This can be a place to practice a usage of a Ruby module - You can extract the querying logic.

As a rule of thumb - [classes should do one thing](https://en.wikipedia.org/wiki/Single_responsibility_principle) (descriptively, in words) - if you describe  the controller as a class that 'queries, validates, pages and returns restaurants' -  that's not a